test/with_dummyserver/test_chunked_transfer.py

```diff
@@ -65,7 +65,7 @@ def _test_body(self, data: bytes | str | None) -> None:
 
             assert b"Transfer-Encoding: chunked" in header.split(b"\r\n")
             if data:
-                bdata = data if isinstance(data, bytes) else data.encode("utf-8")
+                bdata = data if isinstance(data, bytes) else data.encode("latin-1")
```

Comment:
This is actually suspicious because 1.26.x used utf-8 here:

https://github.com/urllib3/urllib3/blob/3c0148048a523325819377b23fc67f8d46afc3aa/test/with_dummyserver/test_chunked_transfer.py#L53-L75

Reply:
Weird, but there was no test case for non-ASCII characters?

Reply:
There are a few non-ASCII characters on line 75

Reply:
And changing `data.encode("utf-8")` to `data.encode("latin-1")` breaks the test in 1.26.x
```
test_unicode_body failed; it passed 0 out of the required 1 times.
        <class 'AssertionError'>
        assert ((b'\r\n' + b'thisshouldbeonechunk\r\n\xe4\xf6\xfc\xdf') + b'\r\n') in b'1e\r\nthisshouldbeonechunk\r\n\xc3\xa4\xc3\xb6\xc3\xbc\xc3\x9f\r\n0\r\n\r\n'
```

Reply:
Related encoding happens on this line in 1.26.x
https://github.com/urllib3/urllib3/blob/57181d6ea910ac7cb2ff83345d9e5e0eb816a0d0/src/urllib3/connection.py#L274

Reply:
Thanks, good find! So we were encoding chunks with UTF-8 and other bodies with Latin-1, and 2.0 reversed that unintentionally. I'll fix that too.